lib/ansible/modules/cloud/docker/docker_swarm_facts.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/python
+#
+# Copyright 2016 Red Hat | Ansible
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: docker_swarm_facts
+
+short_description: Retrieves facts about docker swarm cluster from swarm manager
+
+description:
+  - Retrieves facts about a docker swarm.
+
+version_added: "2.8"
+
+extends_documentation_fragment:
+    - docker
+
+author:
+    - Piotr Wojciechowski (@wojciechowskipiotr)
+    - Thierry Bouvet (@tbouvet)
+
+
+requirements:
+    - python >= 2.7
+    - "docker >= 2.6.0"
+    - "Please note that the L(docker-py,https://pypi.org/project/docker-py/) Python
+       module has been superseded by L(docker,https://pypi.org/project/docker/)
+       (see L(here,https://github.com/docker/docker-py/issues/1310) for details).
+       Version 2.1.0 or newer is only available with the C(docker) module."
+    - Docker API >= 1.25
+'''
+
+EXAMPLES = '''
+- name: Get info on node
+  docker_swarm_facts:
+  register: result
+
+'''
+
+RETURN = '''
```

Comment:
What does it return when the current Docker daemon is not part of a swarm? (Then this module could also be used to detect whether Docker daemon is part of a swarm.)